### PR TITLE
[MI-2542] Add feature to show description field in the posts

### PR DESCRIPTION
**Pull request by ayusht2810**
[screen-capture (12).webm](https://user-images.githubusercontent.com/100013900/213643812-a129afa6-0d88-4f3f-a1cf-70f6117b4dd3.webm)

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/serializer/servicenow_event.go [hunk 1]
```diff
@@ -50,9 +51,22 @@ func (se *ServiceNowEvent) CreateNotificationPost(botID, serviceNowURL, pluginUR
 	}
 
 	titleLink := fmt.Sprintf("%s/nav_to.do?uri=%s.do%%3Fsys_id=%s%%26sysparm_stack=%s_list.do%%3Fsysparm_query=active=true", serviceNowURL, se.RecordType, se.RecordID, se.RecordType)
+
+	if se.Description == "" {
+		se.Description = "N/A"
```
Comment (raghavaggarwal2308) [inline server/serializer/servicenow_event.go:56]: I think we should create a constant for "N/A"

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/containers/custom_post/sericeNowPost.tsx [hunk 2]
```diff
@@ -3,19 +3,22 @@ import {useDispatch} from 'react-redux';
...
 
+import {ModalIds, RecordTypesSupportingComments, RecordTypesSupportingStateUpdation, TypesContainingLink} from 'src/plugin_constants';
 import {setGlobalModalState} from 'src/reducers/globalModal';
+import Utils from 'src/utils';
 
 import './styles.scss';
-import {ModalIds, RecordTypesSupportingComments, RecordTypesSupportingStateUpdation} from 'src/plugin_constants';
 
-type NotificationPostProps = {
+type ShareRecordPostProps = {
```
Comment (manojmalik20) [inline webapp/src/containers/custom_post/sericeNowPost.tsx:12]:
```suggestion
type ServiceNowPostProps = {
```

File: webapp/src/containers/custom_post/sericeNowPost.tsx [hunk 3]
```diff
@@ -72,13 +80,23 @@ const NotificationPost = ({post}: NotificationPostProps) => {
...
-                    <span>
-                        {field.value}
-                    </span>
+                    {field.title === 'Description' ? (
+                        <span>
+                            {description}
+                        </span>
+                    ) : (
+                        <span>
+                            {(type as string) === 'custom_sn_share' ? Utils.getRecordValueForHeader(field.title as TypesContainingLink, field.value) : (field.value)}
```
Comment (manojmalik20) [inline webapp/src/containers/custom_post/sericeNowPost.tsx:89]: What's the need for checking this condition here? `(type as string) === 'custom_sn_share'`

Comment (ayusht2810) [reply webapp/src/containers/custom_post/sericeNowPost.tsx:89]: we are showing navigable links of some fields when a record is shared and field value when for notification post

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**